### ISSUE TITLE
Add exclusions on font-family-not-ending-with-generic-font-family for icon fonts

### ISSUE THREAD

**Issue by lbndev** [racodond/sonar-css-plugin#88]
Hello,

I recently installed this plugin in my Sonarqube. In my projects I get a few of these violations, all of them when using FontAwesome or Glyphicons to display icons in menus and buttons.

To me these are a false positive, because there is no generic font which would be an acceptable replacement.

Would it be possible to exclude well-known icon fonts from the matcher, so that they dont trigger a violation ?

Thanks

SonarQube : 7.0.0.36138
CSS / SCSS / Less plugin : 3.1

**Comment by racodond** [comment]
Hi @lbndev,

Could you please update to the latest version of the plugin (4.13: https://github.com/racodond/sonar-css-plugin/releases/tag/4.13)?

Could you please provide some code samples?

Thank you

David

**Comment by lbndev** [comment]
Hello David,

In the Sonarqube's marketplace, the most recent version I'm offered to install is 3.1
There is no suggested upgrade to 4.13. Is there a reason for this ?

I'd be willing to upgrade manually, but since the update is not suggested by the marketplace, should I fear any instability/incompatibility with SonarQube 7.0 ?

Maybe I'll just give it a try on a standalone, local installation, that'll be safer. I'll keep you informed.

Until then, here is a code sample (violation on the font-family line) : 
```css
input[type="checkbox"] + label:before {
  outline:none;
  content: '\00a0';
  font-family: 'FontAwesome';
  text-align: center;
}
```
Thank you,
Loïc

**Comment by racodond** [comment]
Hi Loïc,

> In the Sonarqube's marketplace, the most recent version I'm offered to install is 3.1
> There is no suggested upgrade to 4.13. Is there a reason for this ?

See https://github.com/racodond/sonar-css-plugin/issues/26 and https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/sonarqube/tnByTHWrdDg/3gARh_nxBQAJ

> I'd be willing to upgrade manually, but since the update is not suggested by the marketplace, should I fear any instability/incompatibility with SonarQube 7.0 ?

There shouldn't be any issue.

> Until then, here is a code sample (violation on the font-family line)

Yes, a comma-separated list of configurable exclusions for those icon fonts could be added. What would be the default value from your point of view?

Thank you

David

**Comment by lbndev** [comment]
> Yes, a comma-separated list of configurable exclusions for those icon fonts could be added. What would be the default value from your point of view?

IMHO at least FontAwesome + Glyphicons Halflings (Bootstrap) + Ionicons (Ionic framework) + Genericons (Wordpress).

But that's really opinionated, and there are a lot more icon fonts available out there, so the ability to configure the exclusion list really makes sense.

Thanks
